src/java/htsjdk/samtools/util/SimpleInterval.java

```diff
@@ -0,0 +1,54 @@
+package htsjdk.samtools.util;
+
+
+/**
+ * Minimal immutable class representing a 1-based closed ended genomic interval
+ *
+ *@warning 0 length intervals are allowed, which are represented as end = start-1
+ */
+public class SimpleInterval implements Locatable {
+
+    private final int start;
+    private final int end;
+    private final String contig;
+
+    /**
+     * Create a new immutable 1-based interval of the form [start, end]
+     * @param contig the name of the contig
+     * @param start  1-based inclusive start position
+     * @param end  1-based inclusive end position
+     */
+    public SimpleInterval(final String contig, final int start, final int end){
+        if(contig == null){
+            throw new IllegalArgumentException("contig cannot be null");
```

Comment:
`Locatable` interface documentation allows null contig names, though:

     /**
      * Gets the contig name for the contig this is mapped to.  May return null if there is no unique mapping.
      * @return name of the contig this is mapped to, potentially null
      */
     String getContig();